base_multi_company/hooks.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2015-2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import api, SUPERUSER_ID
+
+
+__all__ = [
+    'create_company_assignment_view',
+    'post_init_hook',
+    'uninstall_hook',
+]
+
+
+def create_company_assignment_view(cr):
+    cr.execute("""
+        CREATE OR REPLACE VIEW res_company_assignment
+            AS SELECT id, name, parent_id
+            FROM res_company;
+    """)
+
+
+def set_security_rule(env, rule_ref):
+    """Set the condition for multi-company in the security rule.
+
+    :param: env: Environment
+    :param: rule_ref: XML-ID of the security rule to change.
+    """
+    rule = env.ref(rule_ref)
+    if not rule:  # safeguard if it's deleted
+        return
+    rule.write({
+        'active': True,
+        'domain_force': (
```

Comment:
I am doing https://github.com/OCA/multi-company/pull/136 which has error from this rule writing. In partner_multi_company, it will error when I create a new company which also create res.partner.
Can you change to following, because normally we need to have term "or company_id = False"
```suggestion
        'domain_force': (
            "['|', '|', ('company_ids', 'in', user.company_id.ids),"
            " ('visible_for_all_companies', '=', True), ('company_id', '=', False)]"
        ),

```

Reply:
I think now that a better technique is to duplicate the rule, and deactivate the old one, and on uninstall, delete the new one, and activate the old.

Reply:
Allow me to confirm my understanding,
The domain_force in this **base_multi_company** should stay like this, and the changes per mentioned goes to **partner_multi_company**, correct?

Reply:
Also, this one can be merged soon to fix dependency? :)

Reply:
@pedrobaeza I think the problem is not the modification of the rule, but the value in domain_force that raise an error when it should not, in a specific case.
It is True that it could be cleaner to do it this way (activate/deactivate) but it won't change this particular problem.

@kittiu I checked a little, I am not sure why it fails exactly, and only in this particular case.
It seems the `visible_for_all_companies` field is computed too late...
Anyway, I don't think that changing the domain for all models/modules is the good solution.
Indeed, the changes have been made to improve the poor performance of this rule, and adding `('company_id', '=', False)` once again revert the performance improvments.

So the domain could eventually be changed in partner_multi_company, to fix this very particular problem, but it will have an impact on performance.

Another ideal which could be implemented in this base module is to add a default value True to `visible_for_all_companies` field.
This way, the record rule won't fail during creation if `visible_for_all_companies` is not  computed yet.
And if it is computed, it will get the right value...
@pedrobaeza any comment about this?

Reply:
@florian-dacosta as you mentioned about default=True, I add 2 lines like this, it is ok.
![image](https://user-images.githubusercontent.com/1973598/59736530-39870880-9284-11e9-8b4c-2b5346d0764f.png)

Reply:
@kittiu Actually the second line is not necessary.
I made the change